DRAFT: UIUX 30 setup new frontend

jayjayjayjayjayjay opened the pull request.

Commit:
Create react app in frontend_new
Added topbar draft

Commit:
Configure theme
Remove unnecessary svg

Commit:
Implement Search TextField and Buttons

Commit:
Clean up code in Search

Commit:
Change from js to ts

Commit:
Merge branch 'main' into UIUX-30-setup-new-frontend

Commit:
Change height of search buttons and search bar

Commit:
Change width again to work for mobile

Commit:
add info buttons to search buttons

Commit:
Add info button file

Commit:
Add pre-commit for frontend_new
Ran prettier over frontend_new

Commit:
Fix info spacing issue

Commit:
Pass text to info buttons now works

Commit:
Add component folder
Replace popover with SlideAlert

Commit:
Implement theme change

Commit:
Make new component for front page text
Fix horizontal scroll issue

Commit:
Change search info to info
other small changes

Commit:
Improve theme names
fix search to match themes

Commit:
Fix up theme issues

Commit:
Change background color implementation to match theme

Commit:
Add background to light theme
change theme colors

Commit:
Get request working

Commit:
Change theme colours

ariadev79 committed:
useSearchParams not working

ariadev79 committed:
trying QueryString

MiaRavuri committed:
basic card

MiaRavuri committed:
Simple theme change

MiaRavuri committed:
Fixed Error in branch 51

ariadev79 committed:
basic working URL

ariadev79 committed:
Routes working both ways

Commit:
Enter to Accurate Search Implemented
Changed Associate search to Associative search

ariadev79 committed:
remove old files

ariadev79 committed:
refactor some changes

Commit:
First draft implementation remove frontpageinfo on search

Commit:
Second Draft Implementation
Condition rendering and no transition out

Commit:
Revert back unnecesary changes

ariadev79 committed:
add some validation

Commit:
Remove console.log line

jayjayjayjayjayjay committed:
Merge pull request #17 from EngineersBox/UIUX-46-Enter-to-Search

UIUX-46:Enter to Accurate Search Implemented

jayjayjayjayjayjay committed:
Merge pull request #18 from EngineersBox/UIUX-32-transition-to-search-results

UI ux 32 transition to search results

MiaRavuri committed:
Changed the size of cards

MiaRavuri committed:
Fixed minor issues from PR and reverted back to old colours

Commit:
Merge branch 'UIUX-30-setup-new-frontend' into UIUX-51-Cards

MiaRavuri committed:
card media

MiaRavuri committed:
card media

Commit:
Make cards match theme
Implement jump to external link on click
Centre cards

Commit:
Remove logo file

jayjayjayjayjayjay committed:
Merge pull request #21 from EngineersBox/UIUX-51-search-result-cards

UI ux 51 search result cards

Commit:
Remove draft logo

ariadev79 committed:
fix merge conflict and some refactoring

Commit:
Draft refactor of theme

Commit:
Small changes to theme and search

Commit:
Remove console log

Commit:
Remove redundant code in topbar

Commit:
Resolve issue with cards

Commit:
Add drawer for bookmark from sharons branch
Change size
Remove unnecssary code
Change paper colour

Commit:
Add clickable icon to cards
Bit challenging having the icon button being clickable on ontop of the listitem button

Commit:
Make bookmark color secondary
change default icon to unfilled bookmark

ariadev79 committed:
Merge remote-tracking branch 'origin/UIUX-30-setup-new-frontend' into UIUX-53-Add-Route-Link

ariadev79 committed:
fixing conflict

ariadev79 committed:
fixed conflict!

ariadev79 committed:
simple code to store in local storage

Commit:
Bookmarks saved into state
Cards update to reflect that bookmark has been saved

Commit:
Bookmarks working
Need to clean up cards
Need to clean up drawer

Commit:
Remove console.log statements from app

ariadev79 committed:
store the last colour state

ariadev79 committed:
handle switch icon change

Valentina-H0 committed:
just push the Jjdgment functionn

Commit:
Replace var with let

Valentina-H0 committed:
new change based on Jack's comment

Valentina-H0 committed:
Merge pull request #27 from EngineersBox/UIUX-68-theme-setting-function

Draft: UIUX-68-theme-setting-function

EngineersBox commented:
FYI you have `node_modules` checked in. We should not have that commit to main when this gets merged. Those are generated files so it just becomes bloat.

jayjayjayjayjayjay commented:
Please don't merge branches @EngineersBox

Commit:
Remove node modules

Commit:
Improve apprear of theme switch

Commit:
Change name of customswitch to theme switch
Change colours

Commit:
Merge branch 'UIUX-30-setup-new-frontend' into UIUX-71-bookmark-results

Commit:
Change drawer

Commit:
Add back package-lock

Commit:
Fix up scroll issue
Fix up issue with bookmarks not updating

Commit:
Merge branch 'UIUX-53-Add-Route-Link' into UIUX-30-setup-new-frontend

Commit:
Merge branch 'UIUX-30-setup-new-frontend' into UIUX-77-make-theme-switch-pretty

Commit:
Remove unnecessary package lock

Commit:
Merge branch 'UIUX-30-setup-new-frontend' into UIUX-77-make-theme-switch-pretty

Commit:
Make theme inital state depend on time

Commit:
Colour fixes

ariadev79 committed:
rename the title

Commit:
Drawer under appbar
change where close drawer button is

jayjayjayjayjayjay committed:
Merge pull request #29 from EngineersBox/UIUX-65-Renaming-Website

UIUX-65: Renaming website

MiaRavuri committed:
Modified background colour

MiaRavuri committed:
Removed comments

MiaRavuri committed:
Merged branch

ariadev79 committed:
able to store in localStorage

ariadev79 committed:
still sorting out the id

Commit:
Remove console log line

Commit:
Merge branch 'UIUX-30-setup-new-frontend' into UIUX-71-bookmark-results

ariadev79 committed:
attempt removal

ariadev79 committed:
remove save to local storage for bookmarks for now

ariadev79 committed:
remove dangling functions

Commit:
Add themeswitch

Commit:
Fix up issue with theme change
Change function name from themeSwitch to themeChange to avoid confusion

ariadev79 committed:
trying to mao

Commit:
Remove console log statement

Commit:
Improve code according to PR comments

Commit:
Change times theme switches
Improve Cards code

ariadev79 committed:
Merge remote-tracking branch 'origin/UIUX-30-setup-new-frontend' into UIUX-64-local-storage

ariadev79 committed:
the checks are working but not the colours

Commit:
Fix typescript issues

ariadev79 committed:
try to include the default colour

Commit:
Clean up
Remove some custom theme colors

Commit:
Fix up issue with ThemeSetting

ariadev79 committed:
resolve conflict

ariadev79 committed:
use the right operator

Commit:
Merge branch 'UIUX-81-clean-up-code' of github.com:EngineersBox/NewsAggregator into UIUX-81-clean-up-code

Commit:
Change theme colours for light theme

ariadev79 committed:
add tooltips for the bookmarks and theme icon

MiaRavuri committed:
Changed logo to transparent

ariadev79 committed:
conflict

ariadev79 committed:
resolve conflict

ariadev79 committed:
resolve conflict

Commit:
Remove one search button

Commit:
Remove console log

ariadev79 committed:
resolve conflict

Commit:
Autcomplete working

Commit:
Move info button
Change layout to fix up issues from introducing autocomplete

MiaRavuri committed:
changed top bar to blue

Commit:
Make search icon at end of search input
make the icon clickable and have same function as search
Remove Large search button after first search

MiaRavuri committed:
Merge branch 'UIUX-30-setup-new-frontend' into UIUX-33-Theme

Commit:
Fix up issue where if one clicked on autocomplete, it wouldnt set searchinput to value

MiaRavuri committed:
Changed background colour and card colour

ariadev79 committed:
local storage bookmark initial done

ariadev79 committed:
🚧 Merge local storage bookmark test branch

ariadev79 committed:
✅ merge conflict

ariadev79 committed:
🔥 Simplify Code

MiaRavuri committed:
Logos have been updated

Commit:
Finish up Autocomplete functionality

Commit:
Autocomplete url fix

Commit:
Remove commented out code

ariadev79 committed:
Merge remote-tracking branch 'origin/UIUX-30-setup-new-frontend' into UIUX-78-Tooltip

ariadev79 committed:
🚧 Fixing the theme tooltip

ariadev79 committed:
🐛 Use proper package

ariadev79 committed:
📝 Fix naming of theme

ariadev79 committed:
📝 Better name for card bookmark action

MiaRavuri committed:
Logo changes for dark mode

ariadev79 committed:
Merge remote-tracking branch 'origin/UIUX-30-setup-new-frontend' into UIUX-64-local-storage

MiaRavuri committed:
Removed unused imports frontPageInfo.tsx

Commit:
Merge branch 'UIUX-30-setup-new-frontend' into UIUX-83-Search-Auto-Complete

MiaRavuri committed:
Removed unused imports

ariadev79 committed:
fix the null error

ariadev79 committed:
✅ Fix Bookmarks link

ariadev79 committed:
Merge branch 'local-storage-test' into UIUX-64-local-storage

Commit:
Fix up missing import issue
move info button to right

Commit:
Merge branch 'main' into UIUX-30-setup-new-frontend

Commit:
Fix up issue with card text not wrapping on mobile

Commit:
Remove redundant code
Chang wording on info

Commit:
Move theme settings into settings panel
Some fixes too

Commit:
Typo fix

Commit:
Change theme - theme still not finalised, will accept all changes from Mia's branch when merged

MiaRavuri committed:
Changed spacing of logos and size

MiaRavuri committed:
Added logos to search results page

ariadev79 committed:
error with mapping

ariadev79 committed:
fix mapping

ariadev79 committed:
fix errors

ariadev79 committed:
fix function comments

MiaRavuri committed:
Linked logos to go back

Commit:
Remove console log statement

MiaRavuri committed:
Fixed double logo issue on front page

MiaRavuri committed:
Fixed lingering text error

ariadev79 committed:
testing

ariadev79 committed:
Merge branch 'UIUX-33-Theme' of https://github.com/EngineersBox/NewsAggregator into UIUX-33-Theme

MiaRavuri committed:
Unlinked the logos on frontpage

ariadev79 committed:
link back to home page

ariadev79 committed:
pull latest changes

ariadev79 committed:
Merge branch '33-link' into UIUX-33-Theme

ariadev79 committed:
✨ Fix the link to main page

MiaRavuri committed:
removed unncecessary code

MiaRavuri committed:
Merge branch 'UIUX-33-Theme' of https://github.com/EngineersBox/NewsAggregator into UIUX-33-Theme

ariadev79 committed:
fix type issues

ariadev79 committed:
remove comments

Commit:
Make function for bookmark drawer icon

MiaRavuri committed:
Changed padding

Commit:
Change indendation for get

Commit:
Merge branch 'UIUX-64-local-storage' into UIUX-30-setup-new-frontend

Commit:
Merge branch 'UIUX-83-Search-Auto-Complete' into UIUX-30-setup-new-frontend

MiaRavuri committed:
changed spacing

Commit:
Merge branch 'UIUX-30-setup-new-frontend' into UIUX-33-Theme

MiaRavuri committed:
Updated info text and card text colour

MiaRavuri committed:
Changed the icon of the tab to our logo

MiaRavuri committed:
Improved clarity of logos

MiaRavuri committed:
Darkened background colour of light mode

MiaRavuri committed:
Changed the logo size and margin of the logo

MiaRavuri committed:
finalised colours and logo

MiaRavuri committed:
Changed dimensions of the logo so it doesnt get squished

MaddyRav committed:
deleted VS code file c_cpp_properties.json

MaddyRav committed:
Removed launch.json file from the VS code files

MaddyRav committed:
Deleted settings.json file from VS code files

MaddyRav committed:
Changed this back to the old version of the file

XuguangSong98 reviewed (dismissed):
when there are no serious bugs, merge it in main

Commit:
Fix no wrap of url
Refactor how colours for cards work

Commit:
Add box to align image

Commit:
Run prettier over all files

XuguangSong98 reviewed (dismissed):
merge approved

Commit:
Add no result prompt

EngineersBox commented on frontend_new/src/components/Res.js:
```diff
@@ -0,0 +1,65 @@
+import React from "react";
```
**[Question]** Pure JS file? What's the necessity here for it as opposed to TS?

EngineersBox commented on frontend_new/src/index.tsx:
```diff
@@ -0,0 +1,17 @@
+// If you want to start measuring performance in your app, pass a function
+// to log results (for example: reportWebVitals(console.log))
+// or send to an analytics endpoint. Learn more: https://bit.ly/CRA-vitals
+reportWebVitals();
```
**[Question]** Do we need it? If it's not going to have a logger passed to it that is.

jayjayjayjayjayjay replied:
can be removed

ariadev79 commented on frontend_new/src/App.tsx:
```diff
@@ -0,0 +1,67 @@
+  };
+
+  const [bookmarks, setBookmarks] = React.useState<{ [key: number]: any }>({});
+  function handlebookmark(
```
**[Question]** Since we are already using the local storage for bookmarks, do we still need this?

ariadev79 commented on frontend_new/src/components/BookmarkDrawer.tsx:
```diff
@@ -0,0 +1,222 @@
+          <Grid item xs />
+          <Settings themeChange={props.themeChange} />
+          <Grid item>
+            <InfoButton text="This is a description" />
```
**[Question]** Why do we need this?

ariadev79 reviewed (commented):
Others were more or less reviewed by the other PRs

Commit:
'Revert' favicon for old ui

Commit:
Merge branch 'main' into UIUX-30-setup-new-frontend

EngineersBox reviewed (approved):
LGTM
